### PR TITLE
fix(data): make dataset splits truthful and evaluation-safe

### PR DESCRIPTION
**Pull request by liq22**
## User problem

Several maintained data paths could look successful while using incorrect split semantics:

- `data_factory` creates validation datasets with `mode="val"`, but `Default_dataset` only split `mode="valid"`, so validation reused all windows including training windows;
- FS/GFS/pretrain reuse the same file IDs for train and test, so the default test dataset reused the same window pool;
- validation and test samplers used `drop_last=True`, which could silently produce zero evaluation batches when a split was smaller than `data.batch_size`;
- signals shorter than `data.window_size` were printed and skipped instead of stopping with a repair instruction.

## First-principles contract

```text
same raw file
→ one deterministic window list
→ non-overlapping train / val / held-out test slices
```

For DG/CDDG paths whose test IDs are distinct, `mode="test"` still uses the complete test file. For FS/GFS/pretrain paths whose current ID selection reuses files, `mode="test"` selects the configured held-out window tail.

## Scope

- normalize `valid` to the public `val` spelling;
- validate train/val/test ratios;
- make random window selection deterministic through optional `data.window_sampling_seed` (default `0`) so separate split objects select the same source window list before slicing;
- keep FS/GFS/pretrain train, validation, and test windows mutually exclusive;
- preserve incomplete validation/test batches with `drop_last=False`;
- fail with the file ID, observed length, and repair action when a signal cannot produce a window;
- simplify sampler selection without changing GFS episodic training behavior.

## Tests

The existing shared classification runtime contract now checks:

- FS same-file train/val/test windows are disjoint;
- `val` and legacy `valid` select the same validation slice;
- an evaluation split smaller than batch size still yields one batch;
- a short signal fails with an actionable message.

The existing offline Dummy demo remains the end-to-end regression gate.

## Explicit non-goals

```text
no reader numerical changes
no cache rewrite
no dataset adapter registry change
no Pipeline 03/04 promotion
no new hash or manifest
no universal dataset schema
no benchmark-performance claim
```

## Review state

Draft pending Core quality gates and review of compatibility impact on existing dataset subclasses.

### CODE REVIEW

**Review by liq22 — COMMENTED** [review]
Compatibility re-review completed on the current head. DG/CDDG keep full-file test behavior because their test IDs are distinct; FS/GFS/pretrain use held-out same-file windows. Validation/test samplers preserve short batches, GFS episodic training remains unchanged, and the complete offline Dummy path passes. No blocking issue remains.
